packages/cactus-test-plugin-htlc-eth-besu/src/test/typescript/integration/plugin-htlc-eth-besu/initialize-endpoint.test.ts

```diff
@@ -0,0 +1,230 @@
+});
+
+test("Test error initialize endpoint", async (t: Test) => {
+  const keychain = new PluginKeychainMemory({
+    backend: new Map([
+      [accountRef, firstHighNetWorthAccount],
+      [privateKeyRef, besuKeyPair.privateKey],
+    ]),
+    keychainId,
+    logLevel,
+    instanceId: uuidv4(),
+  });
+
+  t.comment("Starting Besu Test Ledger");
+  const besuTestLedger = new BesuTestLedger();
+  await besuTestLedger.start();
+
+  test.onFinish(async () => {
+    await besuTestLedger.stop();
+    await besuTestLedger.destroy();
+  });
+
+  const rpcApiHttpHost = await besuTestLedger.getRpcApiHttpHost();
+  const factory = new PluginFactoryLedgerConnector({
+    pluginImportType: PluginImportType.LOCAL,
+  });
+  const connector: PluginLedgerConnectorBesu = await factory.create({
+    rpcApiHttpHost,
+    pluginRegistry: new PluginRegistry(),
+    instanceId: connectorId,
+  });
+
+  const httpServer = createServer();
+  await new Promise((resolve, reject) => {
+    httpServer.once("error", reject);
+    httpServer.once("listening", resolve);
+    httpServer.listen(0, "127.0.0.1");
+  });
+
+  const addressInfo = httpServer.address() as AddressInfo;
+  t.comment(`HttpServer AddressInfo: ${JSON.stringify(addressInfo)}`);
+  const nodeHost = `http://${addressInfo.address}:${addressInfo.port}`;
+  t.comment(`Cactus Node Host: ${nodeHost}`);
+
+  const configService = new ConfigService();
+  const apiServerOptions = configService.newExampleConfig();
+
+  const pluginRegistry = new PluginRegistry({});
+  pluginRegistry.add(keychain);
+  pluginRegistry.add(connector);
+  const pluginOptions: IPluginHtlcEthBesuOptions = {
+    instanceId: uuidv4(),
+    logLevel,
+    pluginRegistry,
+  };
+
+  const factoryHTLC = new PluginFactoryHtlcEthBesu({
+    pluginImportType: PluginImportType.LOCAL,
+  });
+  const pluginHtlc = await factoryHTLC.create(pluginOptions);
+  pluginRegistry.add(pluginHtlc);
+  apiServerOptions.configFile = "";
+  apiServerOptions.apiCorsDomainCsv = "*";
+  apiServerOptions.apiPort = addressInfo.port;
+  apiServerOptions.cockpitPort = 0;
+  apiServerOptions.apiTlsEnabled = false;
+  const config = configService.newExampleConfigConvict(apiServerOptions);
+  const apiServer = new ApiServer({
+    httpServerApi: httpServer,
+    config: config.getProperties(),
+    pluginRegistry,
+  });
+
+  await apiServer.start();
+  t.comment("Start server");
+  test.onFinish(() => apiServer.shutdown());
+
+  const configuration = new Configuration({
+    basePath: nodeHost,
+  });
+  t.comment("Setting configuration");
+
+  const api = new DefaultApi(configuration);
+  t.comment("Api generated");
+
+  const request: InitializeReq = {
+    keychainId: "fakeId",
+    connectorId,
+    accountRef,
+    privateKeyRef,
+    gas: estimatedGas,
+    credentialType: Web3SigningCredentialType.PRIVATEKEYHEX,
+  };
+  try {
+    const res = await api.initialize(request);
+    t.equal(res.status, 500, "response status is 500");
```

Comment:
@AzaharaC Code 500 should be returned for unknown errors. If the server validated the request and found that it had problems (such as something not being found that the request referred to as something in existence or something being in an invalid format) then a 400 or some other 4xx code could better indicate to end users that they need to fix something in their request.
In general 5xx codes mean that the backend is "at fault" meaning that the developer who sent the request has no action need, but in this case they do (which can be expressed in general with some 4xx codes that in general means that the backend thinks that the request sender is at fault)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
done